### ISSUE TITLE
Adapter to be used with FreeSSM

### ISSUE THREAD

**Issue by jmpolom** [Comer352L/FreeSSM#20]
I read through the readme and a previous issue regarding the type of cable to be used between the PC running FreeSSM and the vehicle. Will one of the commonly available USB to OBDII type adapters work for FreeSSM? [Help with SSM1 adapter](https://github.com/Comer352L/FreeSSM/issues/4) seems to suggest you need a TTL serial adapter to communicate with the vehicle but doesn't really explain the pinout of the OBDII connector that you'd have to attach this to. What kind of cable is required?

**Comment by L1800Turbo** [comment]
As the SSM1 support is in beta state it's unfortunalety not well documented yet. Also the differences between SSM1 and SSM2

In short:
**SSM2** vehicles (mostly from 1998 and upwards) use the commonly used K-Line / OBDII / ... adapters via CARB-Plug (common for OBD-II vehicles).

**SSM1** vehicles are the older ones. You communicate by a TTL serial adapter with the yellow SSM plug.
While we don't have an ordinary documentation you can get the pinout and more information about the interface at http://www.alcyone.org.uk/ssm/ports.html

If you're not sure which protocol you're using let me know about the car and the control unit you want to communicate with.

**Comment by Comer352L** [comment]
Pins 12+13 in the OBD2 connector are used for SSM1 RX+TX.

**Comment by Comer352L** [comment]
Checking the OBD2-connector of the car usually helps to find out which protocols are used (unused pins are missing).

**Comment by gerrievanzyl** [comment]
I am using FreeSSM on a raspberry Pi (compiled for small screens) with the VAG KKL 409.1 USB adapter.  It successfully connects to the ECU of my 05 Impreza STi and allows me to read codes, monitor variables, and reset memory, etc.  However, it fails to connect to any of the other systems, specifically the ABS system (it gives a communication error).  

On the wiring diagrams I see that SAE J1850 protocol is supported on pins 2 and 10.  Do I need a connector that reads these pins to connect to the ABS?

Is it necessary to put the car into diagnostic mode (connect the green plugs) or some other trick to read these subsystems?

(Apologies for using the issue tracking for this question, but it was unclear where else I could ask the question)

**Comment by Comer352L** [comment]
>  am using FreeSSM on a raspberry Pi (compiled for small screens) with the VAG KKL 409.1 USB adapter. It successfully connects to the ECU of my 05 Impreza STi and allows me to read codes, monitor variables, and reset memory, etc. However, it fails to connect to any of the other systems, specifically the ABS system (it gives a communication error).

That's no surprise, because the software currently only supports communication with ABS controllers using the old SSM1 protocol.
You will need a different interface for that (see above).
Later models are using ISO-14230 (KWP2000) (pin 7) instead, which is a different story...
Between ~ 2000 and 2006 both protocols were in use (depending on model, year, market, system manufacturer etc.), but KWP2000 is much more likely to be used in a 2005 model.
Are pins 12+13 present in your OBD-connector ?
If they are missing, you can be sure your ABS uses the new protocol.


> 
> On the wiring diagrams I see that SAE J1850 protocol is supported on pins 2 and 10. Do I need a connector that reads these pins to connect to the ABS?

I have never seen SAE-J1850 beeing used. Are you sure the pin numbering is correct ?
Using a a mirror, pin 2 becomes 7 and 10 becomes 15, which would be ISO9141/14230 K/L-Line pins...

> 
> Is it necessary to put the car into diagnostic mode (connect the green plugs) or some other trick to read these subsystems?

No, that's not required. It doesn't affect the ABS and communication with the controller(s) should always work.

**Comment by digitaltrousers** [comment]
Hi. Do you think this adapter would work: http://www.obdii365.com/wholesale/mini-vci-for-toyota-with-toyota-tis.html (it says it supports "K-Line ISO 9141/KWP 2000 ISO 14230-4/SAE J1850 PWM/SAE J1850 VPW/CAN 2.0B ISO 11898 /CAN ISO 15765-4")
If not, do you have a recommended adapter?
Also can I confirm that I should be able to read airbag codes using the SSM1 pins and a FTDI FT232 TTL adapter and FreeSSM (2006 Outback)?
Thanks.

**Comment by paladinshiva** [comment]
Look on Amazon or eBay for FT232RL K DCAN, any of those will work.

Do not know about this one.

**Comment by mathew1977** [comment]
https://www.ebay.com/itm/303978194439 this one will work.

**Comment by KrisK63** [comment]
I also have a similar question,  but the protocols are different. 
2011 subaru outback cvt.  Attached is the screenshot.  I haven't yet compiled the sw, so cannot comment on that.  I made my own cable (obdii to bb-elec usb2329m)  at the db9.  SP210x at the usb.  I also have a two channel vector cancard xl, if needed.   Thanks
![20230810_083633](https://github.com/Comer352L/FreeSSM/assets/141044435/36fdb3fc-134a-4a7b-878e-ab08802b1357)
![20230810_143836](https://github.com/Comer352L/FreeSSM/assets/141044435/bc837a3f-6509-4868-962a-c33564856933)
![Subaru data link connector pinout](https://github.com/Comer352L/FreeSSM/assets/141044435/8995295f-11df-442b-a84a-d4cc415c0959)

**Comment by Comer352L** [comment]
The description of your interface design is vague / incomplete.

How is the interface connected to your PC and how does the software 
interface look like ?

Which diagnostic protocols does it support ?
For a 2011 CVT, you need an interface with ISO-15765-CAN support.

**Comment by KrisK63** [comment]
Agreed. But you are light years ahead of me.  As mentioned before. My son has totalled two cars in the last year and I'm just worn out.
Basically
What I'm trying to do is simply this:
1.  Obtain a version of freessm that supports my 2011 cvt.  I need to determine if the design ridden failure is too much to fix ( torque converter. Cvt itself, etc).
2.  I'm very familiar with using diagnostic tools,  but everything including the interface (pc to vehicle is provided)
3.  I reinstalled windows 7 x86 for a clean install and ready to kick some ass.

As a side note, I appreciate your help trying to decrypt my scribbling and mess.  Either way,  I have taken your deiect source code and trying to build/comoile/install it for my application.
All I can say is thank you for dealing with my zero knowledge of the subject matter.

Regards,

Kris

________________________________

**Comment by kuri05** [comment]
Thank you, @Comer352L , for putting in the work to maintain a tool like this.  I've done some tool dev for hardware-interfacing software and it can be a really monumental task.

I don't have a question.  This thread answered it already.  I bought a VAG-COM ("Generic OBD2 USB VAG-COM Scan Tool Cable FTDI FT232RL- Subaru Sk...") on ebay for ~$36.  It arrives Monday.  Then we'll see how well it works.

I have to clone an existing ECM for my 2003 Outback, which we apparently toasted while swapping engines?  We're getting <=3V to a few sensors which should have 5V, and it seems to be causing the injectors not to fire.

A mechanic recommended FreeSSM to do the cloning.  If this works, it will put an ends of a month long process of replacing the engine my kids let run out of oil.  Praying for the best.

But I wanted to say thank you.

**Comment by Sleeper85** [comment]
First, thank you to the developer for your work.

I ordered [this VAG KKL with FTDI FT232RL Chip](https://aliexpress.com/item/1005003098261676.html) cable to access the information on my `Subaru Forester 2012` (SSM2 protocol if I understand correctly), but it doesn't work with either `FreeSSM 1.2.5` or `BtSSM 25`.

I select position `1` to connect to `K-LINE` pin `7` at `4800` bps.

At this point, I'm wondering how to verify that this adapter is working properly ?

<img width="500" height="500" alt="Image" src="https://github.com/user-attachments/assets/9132da20-caa0-4f77-b3ad-712cdf49fa59" />

Thank for helping
